### PR TITLE
Add props to TextEditor 

### PR DESCRIPTION
**Pull request by NeenaDev**
Changes - 
Using  `<ControlledEditor />`  instead `<TextEditor/>`
UI fixes to `<ThemePreviewCard />`

### CODE REVIEW

File: library/src/scripts/textEditor/TextEditor.tsx [hunk 1]
```diff
@@ -25,11 +28,12 @@ export default function TextEditor(props: ITextEditorProps) {
             <button onClick={toggleTheme} className={classes.themeToggleIcon} disabled={!isEditorReady}>
                 <ToggleIcon />
             </button>
-            <Editor
+            <ControlledEditor
                 theme={intialTheme}
                 language={language}
                 editorDidMount={handleEditorDidMount}
```
Comment (cchabilall83) [inline library/src/scripts/textEditor/TextEditor.tsx]: do we still need editorDidMount prop?

Comment (NeenaDev) [reply library/src/scripts/textEditor/TextEditor.tsx]: yes, if we remove this, it affect theme toggle button event.

File: library/src/scripts/textEditor/textEditorStyles.ts [hunk 2]
```diff
@@ -32,13 +40,25 @@ export const textEditorClasses = useThemeCache(() => {
             height: viewHeight(90),
             backgroundColor: theme === "dark" ? colorOut("#1f2024") : colorOut(globalVars.elementaryColors.white),
             position: "relative",
+
+            $nest: {
+                "&& .monaco-editor": {
+                    paddingTop: unit(vars.editorPadding.padding.top),
+                    paddingLeft: unit(vars.editorPadding.padding.left),
```
Comment (charrondev) [inline library/src/scripts/textEditor/textEditorStyles.ts]: I'm pretty sure this is the cause of the editor menu being offset.

You'll need to add it to a container further outside the editor root like in the example here. https://wangchujiang.com/react-monacoeditor/

**Comment by charrondev** [comment]
@NeenaDev You've also got merge conflicts.

**Pull request merged**